doc/source/security/enabling-master-key.rst

```diff
@@ -0,0 +1,214 @@
+.. _enabling-master-key:
+
+==============================================================================
+Enabling Master Key Encryption
+==============================================================================
+
+.. _enabling-vault:
+
+Enabling the Keyring Vault
+---------------------------
+
+To enable Master key vault encryption, the user must have
+`SYSTEM_VARIABLES_ADMIN
+<https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_system-variables-admin>`__
+privileges.
+
+Add the following statements to my.cnf:
+
+.. code-block:: MySQL
+
+    [mysqld]
+    early-plugin-load="keyring_valut=keyring_valut.so"
+    loose-keyring_value_config="/home/mysql/keyring_vault.conf"
+
+Restart MySQL.
+
+To change the default value, use the following syntax:
+
+.. code-block:: guess
+
+   mysql> SET innodb_default_encryption_key_id = NEW_ID
+
+Here, **NEW_ID** is an unsigned 32-bit integer.
+
+
+Encrypting a General Tablespace
+-------------------------------------------
+
+.. _enabling-file-per-table:
+
+The InnoDB system tablespace is encrypted by using master key encryption. The
+server must be started with the ``--bootstrap`` option.
+
+If the variable :variable:`innodb_sys_tablespace_encrypt` is set to ON and the
+server has been started in the bootstrap mode, you may create an encrypted table
+as follows:
+
+.. code-block:: guess
+
+    mysql> CREATE TABLE t1 (id INT) ENCRYPTION='Y';
+
+To encrypt a file-per-table tablespace, add the `ENCRYPTION` option.
+
+.. code-block:: MySQL
+
+    mysql> ALTER TABLE t1 ENCRYPTION='Y';
+
+You can also disable encryption for a file-per-table tablespace, set the
+encryption to `N`.
+
+.. code-block:: MySQL
+
+    mysql> ALTER TABLE t1 ENCRYPTION='N';
+
+
+.. _encrypting-system-tablespace:
+
+Encrypting System Tablespace
+------------------------------
+
+:Availability: This feature is **Experimental**.
+
+To enable system tablespace encryption, edit the my.cnf file with the following:
+
+* Add ``innodb_sys_tablespace_encrypt``
+* Edit the ``innodb_sys_tablespace_encrypt`` value to "ON"
+
+System tablespace encryption can only be enabled with the ``--initialize``
+option. You cannot encrypt existing tables in the System tablespace.
+.. _encrypting-undo-log:
+
+Encrypting the Undo Log Data
+------------------------------
+
+:Availability: This feature is **Experimental**.
+
+You enable encryption for an undo log data by adding the following line to the
+my.cnf file:
+
+.. code-block:: MySQL
+
+  [mysqld]
+  innodb_undo_log_encrypt=ON
+
+.. _encrypting-binary-logs:
+
+Encrypting Binary Logs
+-------------------------
+
+Enable encryption of binary logs by running the following statement:
+
+.. code-block:: MySQL
+
+    binlog_encryption=ON
+
+You can rotate the encryption key used by |Percona Server| by running the
+following statement:
+
+.. code-block:: MySQL
+
+    mysql> SELECT rotate_system_key("percona_binlog");
```

Comment:
This should be marked as experimental - wasn't it marked experimental previously ?

Reply:
added experimental.

Reply:
This is/was supposed to be GA in 5.7 as it is now 'inherited' from MySQL 8.0 and backported.

Reply:
@georgelorchpercona rotate_system_key is part of our encryption keys rotation, so it was not inherited from upstream.

Reply:
@patrickbirch @georgelorch I see - you added binlog encryption as experimental - I meant just You can rotate the encryption (...) SELECT rotate_system_key("percona_binlog") <- this is still experimental.

Reply:
yes, I was speaking specifically to binlog encryption, not rotation

Reply:
Moved the Experimental tag to the rotate_system_key for the binlog.